[release] 1.0.0-rc.0 (#3402)
mui/base-ui: docs/src/app/(public)/(content)/react/overview/releases/page.mdx

<Subtitle>Changelogs for each Base UI release.</Subtitle>
<Meta name="description" content="Changelogs for each Base UI release." />

## v1.0.0-rc.0

_Dec 4, 2025_

### Autocomplete

- Fix `keepHighlight` focus sync ([#3399](https://github.com/mui/base-ui/pull/3399))

### Combobox

- **Breaking change:** Sync highlight props.
  Removed the `keepHighlight` prop, given the common use cases for it are often command palette related rather than Select related.
  ([#3377](https://github.com/mui/base-ui/pull/3377))

### Menu

> - Sync highlight props ([#3377](https://github.com/mui/base-ui/pull/3377))

Reviewer: ```suggestion
- Add `highlightItemOnHover` prop ([#3377](https://github.com/mui/base-ui/pull/3377))
```

### Number Field

- Ensure hidden input participates in form validation ([#3374](https://github.com/mui/base-ui/pull/3374))
- Improve symbol replacement logic ([#3376](https://github.com/mui/base-ui/pull/3376))
- Fix fractional step snapping ([#3375](https://github.com/mui/base-ui/pull/3375))
- Fix closing after pointer lock exit in Firefox ([#3379](https://github.com/mui/base-ui/pull/3379))
- Fix parsing numbers with Swiss locale ([#3361](https://github.com/mui/base-ui/pull/3361))

### Popover

- Fix modal backdrop on touch ([#3383](https://github.com/mui/base-ui/pull/3383))
- Fix popover glitching when flipped ([#3364](https://github.com/mui/base-ui/pull/3364))

### Select

- Sync highlight props ([#3377](https://github.com/mui/base-ui/pull/3377))

### Tabs

- **Breaking change:** Fix Panel `keepMounted` behavior.

>   The `value` prop is now required on Tab and TabPanel parts.

Reviewer: ```suggestion
  The `value` prop is now required on `<Tabs.Tab>` and `<Tabs.Panel>`.
```

  ([#3372](https://github.com/mui/base-ui/pull/3372))
- Fix omitted `value` behavior ([#3373](https://github.com/mui/base-ui/pull/3373))

### Toast

- Recalculate content height when layout size is fixed ([#3359](https://github.com/mui/base-ui/pull/3359))

### Tooltip

- Improve contained triggers performance ([#3385](https://github.com/mui/base-ui/pull/3385))

> - Fix multiple swipe directions on same axis ([#3392](https://github.com/mui/base-ui/pull/3392))

Reviewer: This goes under Toast

<Subtitle>Changelogs for each Base UI release.</Subtitle>
<Meta name="description" content="Changelogs for each Base UI release." />

## v1.0.0-rc.0

**Dec 4, 2025**

### General changes

> - Fix missing use client directives ([#3408](https://github.com/mui/base-ui/pull/3408))

Reviewer: ```suggestion
- Fix missing `'use client'` directives ([#3408](https://github.com/mui/base-ui/pull/3408))
```

### Autocomplete

- Fix `keepHighlight` focus sync ([#3399](https://github.com/mui/base-ui/pull/3399))

### Checkbox

> - Match native off state in form submission ([#3406](https://github.com/mui/base-ui/pull/3406))

Reviewer: ```suggestion
- Match native unchecked state in form submission ([#3406](https://github.com/mui/base-ui/pull/3406))
```
Other one makes sense for Switch

- Fix closing after pointer lock exit in Firefox ([#3379](https://github.com/mui/base-ui/pull/3379))

### Menu

- Add `highlightItemOnHover` prop ([#3377](https://github.com/mui/base-ui/pull/3377))
- Do not import client components from MenuStore ([#3409](https://github.com/mui/base-ui/pull/3409))

### Number Field

- Ensure hidden input participates in form validation ([#3374](https://github.com/mui/base-ui/pull/3374))
- Improve symbol replacement logic ([#3376](https://github.com/mui/base-ui/pull/3376))
- Fix fractional step snapping ([#3375](https://github.com/mui/base-ui/pull/3375))
- Fix parsing numbers with Swiss locale ([#3361](https://github.com/mui/base-ui/pull/3361))
- Fix pointer lock release when soft clicking in Firefox ([#3378](https://github.com/mui/base-ui/pull/3378))

### Popover

- Close when pressing focusable element outside ([#3380](https://github.com/mui/base-ui/pull/3380))
- Fix modal backdrop on touch ([#3383](https://github.com/mui/base-ui/pull/3383))
- Fix popover glitching when flipped ([#3364](https://github.com/mui/base-ui/pull/3364))

### Select

- Add `highlightItemOnHover` prop ([#3377](https://github.com/mui/base-ui/pull/3377))

### Switch

> - **Breaking change:** Match native off state in form submission.

Reviewer 2: ```suggestion
- **Breaking change:** Match native off state in form submission.<br />
```
